aws/src/main/java/org/apache/iceberg/aws/AwsProperties.java

```diff
@@ -1070,11 +1124,25 @@ public Map<String, String> httpClientProperties() {
    *     S3Client.builder().applyMutation(awsProperties::applyS3CredentialConfigurations)
    * </pre>
    */
-  public <T extends S3ClientBuilder> void applyS3CredentialConfigurations(T builder) {
+  public <T extends AwsClientBuilder> void applyS3CredentialConfigurations(T builder) {
     builder.credentialsProvider(
         credentialsProvider(s3AccessKeyId, s3SecretAccessKey, s3SessionToken));
   }
 
+  /**
+   * Configure the credentials for AWS clients.
+   *
+   * <p>Sample usage:
+   *
+   * <pre>
+   *     S3Client.builder().applyMutation(awsProperties::applyCredentialConfigurations)
+   * </pre>
+   */
+  public <T extends AwsClientBuilder> void applyCredentialConfigurations(T builder) {
```

Comment:
nit: can we name this `applyClientCredentialConfigurations` to be specific that it's for all clients

Reply:
@dpaani sorry I think this comment is still not addressed, apart from that I am all good, really appreciate the work!

Reply:
hi @jackye1995 sorry I missed it. Can you please check again. thanks

Reply:
Looks like it's not updated?

Reply:
hi @jackye1995 is this what you are expecting to see? This change already committed. Could you please advice if different. thanks

```
/**
   * Configure the credential provider for AWS clients.
   *
   * <p>Sample usage:
   *
   * <pre>
   *     DynamoDbClient.builder().applyMutation(awsProperties::applyCredentialConfigurations)
   * </pre>
   */
  public <T extends AwsClientBuilder> void applyCredentialConfigurations(T builder) {
    if (!Strings.isNullOrEmpty(this.clientCredentialsProvider)) {
      builder.credentialsProvider(credentialsProvider(this.clientCredentialsProvider));
    }
  }
```

Reply:
I am suggesting the method name to be `applyClientCredentialConfigurations` instead of `applyCredentialConfigurations`, which has one more word **Client**

Reply:
my bad :( @jackye1995 . I just fixed it. Thanks jack.